generator/structs.go

```diff
@@ -490,13 +489,13 @@ type GenApp struct {
 	ExtraSchemes        []string
 	Consumes            GenSerGroups
 	Produces            GenSerGroups
-	SecurityDefinitions []GenSecurityScheme
+	SecurityDefinitions GenSecuritySchemes
 	Models              []GenDefinition
 	Operations          GenOperations
 	OperationGroups     GenOperationGroups
 	SwaggerJSON         string
-	// this is important for when the generated server adds routes
-	// ideally this should be removed after we code-generate the router instead of relying on runtime
+	// This is important for when the generated server adds routes.
+	// TODO: ideally this should be removed after we code-generate the router instead of relying on runtime
```

Comment:
There is a very distinct advantage to having that in runtime instead of as generated code.  So I don't think we should ever code generate the router.
If embedding the spec is an issue then it's probably a better solution to use a middleware that loads the files from some other medium (like disk, cdn, ...) and serves that as a static middleware and not embed the specs with `--exclude-spec`